sdk/identity/azure-identity/azure/identity/_credentials/workload_identity.py

```diff
@@ -33,18 +36,28 @@ class WorkloadIdentityCredential(ClientAssertionCredential, TokenFileMixin):
     See the `workload identity overview <https://learn.microsoft.com/azure/aks/workload-identity-overview>`_
     for more information.
 
-    :param str tenant_id: ID of the application's Azure Active Directory tenant. Also called its "directory" ID.
-    :param str client_id: The client ID of an Azure AD app registration.
-    :param str file: The path to a file containing a Kubernetes service account token that authenticates the identity.
+    :keyword str tenant_id: ID of the application's Azure Active Directory tenant. Also called its "directory" ID.
+    :keyword str client_id: The client ID of an Azure AD app registration.
+    :keyword str file: The path to a file containing a Kubernetes service account token that authenticates the identity.
     """
 
     def __init__(
             self,
-            tenant_id: str,
-            client_id: str,
-            file: str,
+            *,
+            tenant_id: Optional[str] = None,
+            client_id: Optional[str] = None,
+            file: Optional[str] = None,
             **kwargs: Any
     ) -> None:
+        tenant_id = tenant_id or os.environ.get(EnvironmentVariables.AZURE_TENANT_ID)
+        client_id = client_id or os.environ.get(EnvironmentVariables.AZURE_CLIENT_ID)
+        file = file or os.environ.get(EnvironmentVariables.AZURE_FEDERATED_TOKEN_FILE)
+        if not tenant_id:
+            raise ValueError("tenant_id is required")
```

Comment:
I feel like the error messages could be more descriptive.

How about for each something like

```python
        raise ValueError(
            "'tenant_id' is required. Please pass it in or set the "
            f"{EnvironmentVariables.AZURE_TENANT_ID} environment variable"
        )
        raise ValueError(
            "'client_id' is required. Please pass it in or set the "
            f"{EnvironmentVariables.AZURE_CLIENT_ID} environment variable"
        )
        raise ValueError(
            "'file' is required. Please pass it in or set the "
            f"{EnvironmentVariables.AZURE_FEDERATED_TOKEN_FILE} environment variable"
        )
```